src/overlays/actors/ovl_Obj_Dora/z_obj_dora.c

```diff
@@ -98,34 +152,188 @@ static DamageTable D_80B61488 = {
+    this->gongAngle.x += 0x1555;
+    this->gongAngle.z += 0x238E;
+    this->gongRotation.x = Math_SinS(this->gongAngle.x) * this->gongForce.y;
+    this->gongRotation.z = Math_SinS(this->gongAngle.z) * this->gongForce.z;
+}
+
+bool ObjDora_IsHalfHourly(u16 time) {
+    f32 temp = time;
+    temp -= (1365.3334f * (s32)(time / 1365.3334f));
+
+    if ((temp < 45.511112f) || (1319.8223f < temp)) {
+        return true;
+    }
+
+    return false;
+}
+
+void ObjDora_CheckCollision(ObjDora* this, GlobalContext* globalCtx) {
+    Actor* itemDrop;
+    u16 time;
+
+    if (this->colliderTris.base.acFlags & AC_HIT) {
+        time = gSaveContext.save.time;
+        this->colliderTris.base.acFlags &= ~AC_HIT;
+        this->collisionCooldownTimer = 5;
+
+        switch (this->actor.colChkInfo.damageEffect) {
+            case 14:
+            case 15:
+                if (this->actor.colChkInfo.damageEffect == 15) {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_DORA_S);
+                    this->lastGongHitType = DORA_HIT_LIGHT;
+                } else {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_DORA_L);
+                    this->lastGongHitType = DORA_HIT_STRONG;
+                }
+
+                func_800BC848(&this->actor, globalCtx, 5, 10);
+                ObjDora_SetupMoveGong(this);
+
+                if ((ObjDora_IsHalfHourly(time) == 1) && (this->rupeeDropTimer == 0)) {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_TRE_BOX_APPEAR);
+                    itemDrop = Item_DropCollectible(globalCtx, &this->actor.world.pos, ITEM00_RUPEE_BLUE);
+                    itemDrop->world.rot.y = this->actor.world.rot.y;
+                    itemDrop->world.rot.y += (s32)(Rand_Centered() * 90.0f * 182.04445f);
+                    itemDrop->velocity.y = 5.0f;
+                    itemDrop->gravity = -1.0f;
+                    this->rupeeDropTimer = 40;
+                }
+                break;
+        }
+    }
 
-#endif
+    if (this->rupeeDropTimer > 0) {
+        this->rupeeDropTimer -= 1;
+    }
 
-extern ColliderTrisElementInit D_80B61310[6];
-extern ColliderTrisInit D_80B61478;
-extern DamageTable D_80B61488;
-extern CollisionCheckInfoInit2 D_80B614A8;
+    if (this->collisionCooldownTimer > 0) {
+        this->collisionCooldownTimer -= 1;
+    } else {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderTris.base);
+    }
+}
 
-extern UNK_TYPE D_06004160;
+void ObjDora_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDora* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Dora/ObjDora_Init.s")
+    this->actionFunc(this, globalCtx);
+    ObjDora_CheckCollision(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Dora/ObjDora_Destroy.s")
+void ObjDora_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDora* this = THIS;
+    f32 gongXForce;
```

Comment:
I might have called this `gongForceX` to match the struct variable name order, that's just my opinion though.

Reply:
Looks better I'll change it over.